beacon-chain/sync/data_columns_reconstruct.go

```diff
@@ -49,8 +50,13 @@ func (s *Service) reconstructSaveBroadcastDataColumnSidecars(
 
 	// Retrieve our local node info.
 	nodeID := s.cfg.p2p.NodeID()
-	custodyGroupCount := s.cfg.custodyInfo.ActualGroupCount()
-	localNodeInfo, _, err := peerdas.Info(nodeID, custodyGroupCount)
+	custodyGroupCount, err := s.cfg.p2p.CustodyGroupCount()
+	if err != nil {
+		return errors.Wrap(err, "custody group count")
+	}
+
+	samplingSize := max(custodyGroupCount, samplesPerSlot)
```

Comment:
should we just have a function that returns the sampling size? i saw this in a few places

Reply:
Yes it was my first intention as well.

There is already this function:
https://github.com/OffchainLabs/prysm/blob/0e7bd72b13f9d9ea71e9b5653f9b39c99cac8016/beacon-chain/sync/subscriber.go#L647-L666

which computes the "theoretical" sampling size, which is used at multiple places.

The other computations of `samplingSize` are "actual" sampling size and in that case the `custodyGroupCount` comes from different locations.
==> I can create corresponding function**s**, but they will be called only once.